docs/ispc.rst

```diff
@@ -3999,6 +4000,16 @@ for an optimized release build), use the ``--opt=disable-assertions``
 command-line argument.
 
 
+Compiler Optimization Hints
+---------------------------
+
+The ``ispc`` standard library includes a mechanism for adding ``assume()``
+statements to ``ispc`` program code. The ``assume()`` function takes a
+single uniform boolean expression as an argument. This expression is
+assumed to be ``true`` and this information will be used for optimization
+when possible. This feature is currently only available on CPU targets.
```

Comment:
Other points to make here:
- code generation should not be affected by `assume()` and not runtime checks are done, the information is used for optimizations only.
- list of examples that should work.

Reply:
1. 'code generation should not be affected by assume()' - this might be a confusing statement. Code generation is affected if optimization happens, right? Do you think "code generation should not be aborted by assume and no runtim..." works better?
2. Will add snippets from lit test.

Reply:
Right, this might be confusing. "The condition used in `assume()` statement will not be code generated and does not imply runtime checks. It will solely be used as optimization hint, if compiler is able to use this information."

Reply:
Done